mpi_learn/mpi/process.py

```diff
@@ -194,7 +191,16 @@ def recv(self, obj=None, tag=MPI.ANY_TAG, source=None, buffer=False, status=None
             source = self.parent_rank 
         tag_num = self.lookup_mpi_tag(tag)
         if buffer:
-            comm.Recv( obj, source=source, tag=tag_num, status=status )
+            if type(obj) == list:
```

Comment:
This looks hard to maintain because the Send logic needs to be identical to the Recv logic.  Can we template this so Send and Recv both run the same set of loops?

Reply:
I tried to put the send and receive on the same footing indeed. I'll try the templating next.